src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestActor.cs

```diff
@@ -35,9 +80,71 @@ public PullRequestActor(
     /// <returns>
     ///     True if updates have been applied; <see langword="false" /> otherwise.
     /// </returns>
-    public Task<bool> ProcessPendingUpdatesAsync(SubscriptionUpdateWorkItem update)
+    public async Task<bool> ProcessPendingUpdatesAsync(SubscriptionUpdateWorkItem update)
     {
-        throw new NotImplementedException();
+        _logger.LogInformation("Processing pending updates for subscription {subscriptionId}", update.SubscriptionId);
+
+        // Check if we have an on-going PR for this actor
+        InProgressPullRequest? pr = await _pullRequestState.TryGetStateAsync();
+
+        bool canUpdate;
+        if (pr == null)
+        {
+            _logger.LogInformation("No existing pull request state found");
+            canUpdate = true;
+        }
+        else
+        {
+            canUpdate = await SynchronizeInProgressPullRequestAsync(pr);
+        }
+
+        // Code flow updates are handled separetely
+        if (update.SubscriptionType == SubscriptionType.DependenciesAndSources)
+        {
+            if (pr != null && !canUpdate)
```

Comment:
Good catch. This method changed a bit because `SynchronizeInProgressPullRequestAsync` used to return a tuple `(canUpdate, pr)` so I guess I didn't check for these things.